packages/shared/components/molecules/AssetIcon.svelte

```diff
@@ -22,7 +22,7 @@
 
     $: $network, chainId, (chainName = getTooltipText())
     $: isAnimation = asset.id in ANIMATED_TOKEN_IDS
-    switch (asset.id) {
+    $: switch (asset.id) {
```

Comment:
having such large reactive logic to me is a bit confusing to read, i lose context based on what it triggers
I would make it inside a function

```ts
$: asset, updateAssetIcon()

function updateAssetIcon(): void {
    switch (asset.id) {
        case String(COIN_TYPE[NetworkId.Iota]):
            assetInitials = ''
            assetIconBackgroundColor = '#6E82A4'
            assetIconColor = isBright(assetIconBackgroundColor) ? 'gray-800' : 'white'
            icon = NETWORK_ICON_SVG[NetworkId.Iota]
            break
        case String(COIN_TYPE[NetworkId.Shimmer]):
        case String(COIN_TYPE[NetworkId.Testnet]):
            assetInitials = ''
            assetIconBackgroundColor = '#25DFCA'
            assetIconColor = isBright(assetIconBackgroundColor) ? 'gray-800' : 'white'
            icon = NETWORK_ICON_SVG[NetworkId.Shimmer]
            break
        default:
            assetInitials = getAssetInitials(asset)
            assetIconBackgroundColor = getIconColorFromString(asset.metadata?.name, {
                shades: ['500', '600', '700', '800'],
                colorsToExclude: ['gray'],
            })
            assetIconColor = isBright(assetIconBackgroundColor) ? 'gray-800' : 'white'
            assetLogoUrl = asset.metadata?.standard === TokenStandard.Irc30 ? asset.metadata?.logoUrl ?? '' : ''
            icon = null
    }
}
```